### PR TITLE
Add releases to /api/packages endpoint

### PR DESCRIPTION
**Pull request by enilsen16**
#275 I removed the n+1 query, and am now querying it all in one go.

### CODE REVIEW

File: mix.lock [hunk 1]
```diff
@@ -1,28 +1,30 @@
...
+  "plug": {:hex, :plug, "1.1.2"},
   "poison": {:hex, :poison, "1.5.2"},
   "poolboy": {:hex, :poolboy, "1.5.1"},
   "porcelain": {:hex, :porcelain, "2.0.1"},
-  "postgrex": {:hex, :postgrex, "0.11.0"},
+  "postgrex": {:hex, :postgrex, "0.11.1"},
   "ranch": {:hex, :ranch, "1.2.1"},
+  "ssl_verify_fun": {:hex, :ssl_verify_fun, "1.1.0"},
   "ssl_verify_hostname": {:hex, :ssl_verify_hostname, "1.0.5"},
   "sweet_xml": {:hex, :sweet_xml, "0.6.1"}}
```
Comment (ericmj) [inline mix.lock]: Please do not update all dependencies.

File: web/controllers/api/package_controller.ex [hunk 2]
```diff
@@ -7,9 +7,16 @@ defmodule HexWeb.API.PackageController do
...
     search   = HexWeb.Utils.safe_search(params["search"])
     sort     = HexWeb.Utils.safe_to_atom(params["sort"] || "name", @sort_params)
-    packages = Package.all(page, 100, search, sort) |> HexWeb.Repo.all
+
+    packages = Package.all(page, 100, search, sort)
+      |> HexWeb.Repo.all 
+      |> HexWeb.Repo.preload([:releases])
 
     when_stale(conn, packages, [modified: false], fn conn ->
+      packages = Enum.map(packages, fn(package) ->
```
Comment (ericmj) [inline web/controllers/api/package_controller.ex]: No parenthesis around `package`.

**Comment by ericmj** [comment]
Can you add tests and make sure that when we render the releases we only include the version and the API link to the release.

File: web/controllers/api/package_controller.ex [hunk 3]
```diff
@@ -7,9 +7,16 @@ defmodule HexWeb.API.PackageController do
     page     = HexWeb.Utils.safe_int(params["page"])
     search   = HexWeb.Utils.safe_search(params["search"])
     sort     = HexWeb.Utils.safe_to_atom(params["sort"] || "name", @sort_params)
-    packages = Package.all(page, 100, search, sort) |> HexWeb.Repo.all
+
+    packages = Package.all(page, 100, search, sort)
+      |> HexWeb.Repo.all 
+      |> HexWeb.Repo.preload([:releases])
```
Comment (wojtekmach) [inline web/controllers/api/package_controller.ex]: minor style comment, I think this should be something like:

```elixir
packages =
  Package.all(...)
  |> HexWeb.Repo.all
  |> HexWeb.Repo.preload(...)
```

**Comment by enilsen16** [comment]
@ericmj Let me know if that test suffices. The idea is to check that each map in the releases list only contains 2 elements and those two elements are both the version and the release show url.

**Comment by ericmj** [comment]
Looks great @enilsen16. If you can just clean up the commit that updates httpoison so that it says something like "Update httpoison dependency" then we are ready to merge.

**Comment by enilsen16** [comment]
Awesome.

**Pull request merged**